near-contract-standards/src/event.rs

```diff
@@ -0,0 +1,299 @@
+    {
+        Self {
+            owner_id: owner_id.into(),
+            token_ids: token_ids.into_iter().map(|s| s.into()).collect(),
+            memo: memo.map(|t| t.into()),
+        }
+    }
+}
+
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NftTransferData<'a> {
+    #[serde(borrow)]
+    pub old_owner_id: Cow<'a, str>,
+    #[serde(borrow)]
+    pub new_owner_id: Cow<'a, str>,
+    #[serde(borrow)]
+    pub token_ids: Vec<Cow<'a, str>>,
+    #[serde(borrow)]
+    pub authorized_id: Option<Cow<'a, str>>,
+    #[serde(borrow)]
+    pub memo: Option<Cow<'a, str>>,
+}
+
+impl<'a> NftTransferData<'a> {
+    pub fn new<S>(
+        old_owner_id: S,
+        new_owner_id: S,
+        token_ids: Vec<S>,
+        authorized_id: Option<S>,
+        memo: Option<S>,
+    ) -> NftTransferData<'a>
+    where
+        S: Into<Cow<'a, str>>,
+    {
+        Self {
+            authorized_id: authorized_id.map(|t| t.into()),
+            old_owner_id: old_owner_id.into(),
+            new_owner_id: new_owner_id.into(),
+            token_ids: token_ids.into_iter().map(|s| s.into()).collect(),
+            memo: memo.map(|t| t.into()),
+        }
+    }
+}
+
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NftBurnData<'a> {
+    #[serde(borrow)]
+    pub owner_id: Cow<'a, str>,
+    #[serde(borrow)]
+    pub token_ids: Vec<Cow<'a, str>>,
+    #[serde(borrow)]
+    pub authorized_id: Option<Cow<'a, str>>,
+    #[serde(borrow)]
+    pub memo: Option<Cow<'a, str>>,
+}
+
+impl<'a> NftBurnData<'a> {
+    pub fn new<S>(
+        owner_id: S,
+        token_ids: Vec<S>,
+        authorized_id: Option<S>,
+        memo: Option<S>,
+    ) -> NftBurnData<'a>
+    where
+        S: Into<Cow<'a, str>>,
+    {
+        Self {
+            owner_id: owner_id.into(),
+            token_ids: token_ids.into_iter().map(|s| s.into()).collect(),
+            authorized_id: authorized_id.map(|t| t.into()),
+            memo: memo.map(|t| t.into()),
+        }
+    }
+}
+
+impl<'a> NearEvent<'a> {
+    pub fn new_171(version: &'static str, event_kind: Nep171EventKind<'a>) -> Self {
+        NearEvent::Nep171(Nep171Event { version, event_kind })
+    }
+
+    pub fn new_171_v1(event_kind: Nep171EventKind<'a>) -> Self {
+        NearEvent::new_171("1.0.0", event_kind)
+    }
+
+    pub fn nft_burn(data: Vec<NftBurnData<'a>>) -> Self {
+        NearEvent::new_171_v1(Nep171EventKind::NftBurn(data))
+    }
+    pub fn nft_transfer(data: Vec<NftTransferData<'a>>) -> Self {
+        NearEvent::new_171_v1(Nep171EventKind::NftTransfer(data))
+    }
+
+    pub fn nft_mint(data: Vec<NftMintData<'a>>) -> Self {
+        NearEvent::new_171_v1(Nep171EventKind::NftMint(data))
+    }
+
+    pub(crate) fn to_json_string(&self) -> String {
+        serde_json::to_string(self).unwrap()
+    }
+
+    pub fn to_json_event_string(&self) -> String {
+        format!("EVENT_JSON:{}", self.to_json_string())
+    }
```

Comment:
Let's add [`#[must_use]`](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute) attribute to these methods to make sure users don't forget about emitting the events or at least doing something with the function result.

Consider also adding some text comment `#[must_use = "don't forget to .emit() the event"]` to improve the DX.